tests/test_validate_expected_origin.py

```diff
@@ -0,0 +1,61 @@
+from unittest import TestCase
+
+from webauthn.helpers.validate_expected_origin import (
+    match_origins,
+    validate_expected_origin,
+)
+
+
+class TestValidateExpectedOrigin(TestCase):
+
+
+    def test_match_origins(self):
+        """Test for combined match (exact and wildcard)."""
+        expected_origin_result = [
+            # exact match
+            ("https://www.example.org", "https://www.example.org", True),
+            # wildcard subdomain match
+            ("https://*.example.org", "https://pass.example.org", True),
+            # wildcard prefix-only match
+            ("example-os*", "example-os:appid:204ffa1a5af110ac483f131a1bef8a841a7a", True),
+            # wildcard suffix-only match
+            ("*.com", "https://something-random.com", True),
+            # wildcard match anything - this is a bad idea
+            ("*", "totally random origin that isn't even an origin", True),
```

Comment:
I don't think these should be supported, the secure context that's required for WebAuthn to work requires the origin to be `http://localhost:*` or an `https://` address. Allowing use of a value like `*.com` would allow "http://example.com" which is _not_ a valid domain.

I also don't want to encourage _not_ verifying origins which is why I object to `*` as being a supported value. This should be specifically denied, I'm talking "raise an `Exception`" level of disallow.

Reply:
I'll add an `is_secure_context` function to capture acceptable partials so we don't have such an open barn door.